libs/wingsdk/src/target-sim/service.inflight.ts

```diff
@@ -59,41 +51,57 @@ export class Service implements IServiceClient, ISimulatorResourceInstance {
       return;
     }
 
+    await this.createBundlePromise;
+
+    if (!this.bundle) {
+      this.addTrace("Failed to start service: bundle is not created");
+      return;
+    }
+
+    this.sandbox = new Sandbox(this.bundle.entrypointPath, {
+      env: {
+        ...this.environmentVariables,
+        WING_SIMULATOR_URL: this.context.serverUrl,
+      },
+      log: (internal, _level, message) => {
+        this.addTrace(message, internal);
+      },
+    });
+
     try {
-      this.onStop = await this.sandbox.call("handle");
+      await this.sandbox.call("start");
       this.running = true;
     } catch (e: any) {
-      this.context.addTrace({
-        data: { message: `Failed to start service: ${e.message}` },
-        type: TraceType.RESOURCE,
-        sourcePath: this.context.resourcePath,
-        sourceType: SERVICE_FQN,
-        timestamp: new Date().toISOString(),
-      });
+      this.addTrace(`Failed to start service: ${e.message}`);
     }
   }
 
   public async stop(): Promise<void> {
     // Do nothing if service is already stopped.
-    if (!this.running) {
+    if (!this.running || !this.sandbox) {
       return;
     }
 
-    if (this.onStop) {
-      // wing has a quirk where it will return either a function or an object that implements
-      // "handle", depending on whether the closure is defined in an inflight context or preflight
-      // context. so we need to handle both options here. (in wing this is handled by the compiler).
-      if (typeof this.onStop === "function") {
-        await (this.onStop as any)();
-      } else {
-        await this.onStop.handle();
-      }
+    try {
+      await this.sandbox.call("stop");
+      await this.sandbox.cleanup();
+      this.running = false;
```

Comment:
Maybe set running to false first in case stop is quickly called twice

```suggestion
      this.running = false;
      await this.sandbox.call("stop");
      await this.sandbox.cleanup();
```